airflow/models.py

```diff
@@ -387,9 +387,15 @@ def kill_zombies(self, session=None):
                 dag = self.dags[ti.dag_id]
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
+
+                    # now set non db backed vars on ti
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(str(ti)))
-                    self.log.info('Marked zombie job %s as failed', ti)
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
+
+                    ti.handle_failure("{} detected as zombie".format(ti),
+                                      ti.test_mode, ti.get_template_context())
```

Comment:
Oh - I now see what is going on with the `log.error` vs `log.exception` - the other place handle_failure is used is for a task actually failing, after an exception has been raised. `log.exception` is the same as `log.error(..., exc_info=True)` which just automatically includes exception info.

If this produces sensible logging output now I think this PR looks good now.

Reply:
yeah, the log is giving correct msg, that we `got a zombie` instead of announcing it dead already. As at this point no one is sure what task's handle_failure would be doing, so finally we are stating that after we are pass through it. Here is one such [example](https://github.com/apache/incubator-airflow/blob/master/airflow/contrib/hooks/qubole_hook.py#L98).